pkg/iac/terraform/state/enumerator/s3.go

```diff
@@ -25,8 +26,27 @@ type S3Enumerator struct {
 }
 
 func NewS3Enumerator(config config.SupplierConfig) *S3Enumerator {
+	s3Profile, exist := os.LookupEnv("DCTL_S3_PROFILE")
+	if !exist {
+		s3Profile = os.Getenv("AWS_DEFAULT_PROFILE")
+		if s3Profile == "" {
+			s3Profile = os.Getenv("AWS_PROFILE")
+		}
+	}
+
+	s3Config := aws.Config{}
+
+	if s3Region := os.Getenv("DCTL_S3_REGION"); s3Region != "" {
+		s3Config.Region = aws.String(s3Region)
+		if s3Endpoint := os.Getenv("DCTL_S3_ENDPOINT"); s3Endpoint != "" {
+			s3Config.Endpoint = aws.String(s3Endpoint)
+		}
+	}
+
```

Comment:
Let me suggest a different approach: we could create a struct for those values with a struct tag `env` to define from which environment variable parse the value from. [example from configor](https://github.com/jinzhu/configor#usage)

### Example

```go
type S3EnumeratorConfig struct {
    Endpoint string `env:"DCTL_S3_ENDPOINT"`
    // ...
}

// ...

var s3Config S3EnumeratorConfig
load(&s3Config)

sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
		Profile:           s3Profile,
		Config:            aws.Config{
                Endpoint: aws.String(s3Config.Endpoint),
                // ...
        },
}))
```

This way we could reuse that logic for any other auth system, and easily add new config parameters.

WDYT @cloudskiff/engineering ?

Reply:
I agree with this approach event if I would have preferred something less declarative and more flexible (with this approach we'll have to follow ever AWS SDK evolution). But I didn't get a better approach in mind. Maybe we can do something like in acc test where we override the whole env used to run TF operations ? 

Anyway I don't think this add a lot of value to introduce a config library only for this case, I'd rather prefer to have manual `os.GetEnv` in an `aws.Config` object directly.

Reply:
> with this approach we'll have to follow ever AWS SDK evolution
I don't think we have any other option anyway, do we?

Reply:
So do i try to create a structure like sundowndev recommandation ?

Reply:
> > with this approach we'll have to follow ever AWS SDK evolution
> 
> I don't think we have any other option anyway, do we?

This approach can handle this :

> Maybe we can do something like in acc test where we override the whole env used to run TF operations ?

Reply:
@eliecharra what do you think of this new approach ?